hoomd/md/nlist.py

```diff
@@ -190,235 +176,138 @@ def _detach(self):
+    is size disparity in the cutoff radius. The memory demands of `Stencil` can
+    also be lower than `Cell` if your system is large and has many small cells
+    in it; however, `Tree` is usually a better choice for these systems.
+
+    The performance of `Stencil` depends strongly on the choice of *cell_width*.
+    The best performance is obtained when the cutoff radii are multiples of the
+    *cell_width*, and when the *cell_width* covers the simulation box with a
+    roughly integer number of cells. The *cell_width* must be set manually.
 
     Examples::
 
-        nl_s = nlist.stencil(check_period = 1)
-        nl_s.set_params(r_buff=0.5)
-        nl_s.reset_exclusions([]);
-        nl_s.tune()
-        nl_s.tune_cell_width(min_width=1.5, max_width=3.0)
+        nl_s = nlist.Stencil(cell_width=1.5)
+        nl_s.cell_width = 0.75
+        pair = hoomd.md.pair.LJ(nl_s)
 
-    Note:
-        *d_max* should only be set when slj diameter shifting is required by a pair potential. Currently, slj
-        is the only pair potential requiring this shifting, and setting *d_max* for other potentials may lead to
-        significantly degraded performance or incorrect results.
+    Attributes:
+        cell_width (float): The underlying stencil bin width for the cell list.
+        deterministic (bool): When `True`, sort neighbors to help provide
+            deterministic simulation runs.
     """
-    def __init__(self, r_buff=0.4, check_period=1, d_max=None, dist_check=True, cell_width=None, name=None, deterministic=False):
-        nlist.__init__(self)
-
-        if name is None:
-            self.name = "stencil_nlist_%d" % stencil.cur_id
-            stencil.cur_id += 1
-        else:
-            self.name = name
-
-        # create the C++ mirror class
-        if not hoomd.context.current.device.cpp_exec_conf.isCUDAEnabled():
-            self.cpp_cl = _hoomd.CellList(hoomd.context.current.system_definition)
-            hoomd.context.current.system.addCompute(self.cpp_cl , self.name + "_cl")
-            cls = _hoomd.CellListStencil(hoomd.context.current.system_definition, self.cpp_cl)
-            hoomd.context.current.system.addCompute(cls, self.name + "_cls")
-            self.cpp_nlist = _md.NeighborListStencil(hoomd.context.current.system_definition, 0.0, r_buff, self.cpp_cl, cls)
-        else:
-            self.cpp_cl  = _hoomd.CellListGPU(hoomd.context.current.system_definition)
-            hoomd.context.current.system.addCompute(self.cpp_cl , self.name + "_cl")
-            cls = _hoomd.CellListStencil(hoomd.context.current.system_definition, self.cpp_cl)
-            hoomd.context.current.system.addCompute(cls, self.name + "_cls")
-            self.cpp_nlist = _md.NeighborListGPUStencil(hoomd.context.current.system_definition, 0.0, r_buff, self.cpp_cl, cls)
-
-        self.cpp_nlist.setEvery(check_period, dist_check)
+    def __init__(self, cell_width, buffer=0.4, check_dist=True, deterministic=False,
```

Comment:
We should use the same order of arguments for the common parameters between the different neighbor list classes. Put the parameters that are different at the end of the list. The documentation should be updated to match the order in the signature.